GUIDELINES.md

```diff
@@ -0,0 +1,138 @@
+# Guidelines
+
+This document covers general best-practices when authoring Traits and/or
+Specifications for inclusion in the library.
+
+## Traits
+
+Traits define unique qualities or characteristics of an Entity or
+Relationship.
+
+- Traits must be atomic. No single trait should ever change the
+  meaning of another trait _or its properties_. This ensures that future
+  traits not known at the time code is written can't change the meaning
+  of previously defined values.
+
+- Traits should contain the minimum number of properties required to
+  define a particular characteristic. They must document in their
+  description if properties are optional. Non-optional properties are
```

Comment:
[question] Should trait descriptions distinguish between optional for read, but not for write, or vice versa?